### ISSUE TITLE
transforms v2 rotate can create different sizes between tensors and PIL with expand=True

### ISSUE THREAD

**Issue by pmeier** [pytorch/vision#7714]
```python
import torch
from torchvision.transforms.v2 import functional as F

input_tensor = torch.rand(3, 17, 11)
input_pil = F.to_pil_image(input_tensor)

kwargs = dict(angle=-10.9, center=(1.2, 4.9), expand=True)

output_tensor = F.rotate(input_tensor, **kwargs)
output_pil = F.rotate(input_pil, **kwargs)

shape_tensor = F.get_image_size(output_tensor)
shape_pil = F.get_image_size(output_pil)

assert shape_tensor == shape_pil, f"{shape_tensor} != {shape_pil}"
```

```
AssertionError: [15, 20] != [15, 19]
```

Replacing the import with `from torchvision.transforms import functional as F`, i.e. using transforms v1, does not have this problem.

**Comment by pmeier** [comment]
We had some subtle issues here before, which supposedly should have been fixed by #7271.

**Comment by pmeier** [comment]
PIL kernel is the culprit:

```python
import torch
from torchvision.transforms.v2 import functional as F

input_tensor = torch.rand(3, 17, 11)
input_pil = F.to_pil_image(input_tensor)

kwargs = dict(angle=-10.9, expand=True)

output_w_center = F.rotate(input_tensor, **kwargs, center=(1.2, 4.9))
output_wo_center = F.rotate(input_pil, **kwargs, center=None)

shape_w_center = F.get_image_size(output_w_center)
shape_wo_center = F.get_image_size(output_wo_center)

assert shape_w_center == shape_wo_center, f"{shape_w_center} != {shape_wo_center}"
```

```
AssertionError: [15, 20] != [15, 19]
```

Meaning, #7271 did its job, but we need to expand it to the PIL kernel as well.

The issue is that `center` theoretically should have no effect on the output size if `expand=True` is given. Practically, rounding effects take place which lead the off-by-one errors. The difference between v1 and v2 is that we not only warn users about setting both parameters

https://github.com/pytorch/vision/blob/71968bc4afb8892284844a7c4cbd772696f42a88/torchvision/transforms/v2/functional/_geometry.py#L922-L923

we also explicitly unset `center` in this case

https://github.com/pytorch/vision/blob/71968bc4afb8892284844a7c4cbd772696f42a88/torchvision/transforms/v2/functional/_geometry.py#L924
